core/src/price.ts

```diff
@@ -110,14 +113,18 @@ export const calculateTransactionGrossProfitDate = function (auction: Auction, c
         return undefined;
     }
 
-    let steps = 0;
-    let currentValue = new BigNumber(auction.approximateUnitPrice);
+    const stepNumber = Math.ceil(
+        Math.log(auction.marketUnitPrice.dividedBy(auction.approximateUnitPrice).toNumber()) /
+            Math.log(auction.priceDropRatio.toNumber())
+    );
```

Comment:
Do you think it's possible to use normal `Math` operations over those big numbers? Have you put any thought into it? (I can imagine it may be fine as soon as we fine with the approximate number here, but I haven't though about edge cases, have you?)

Reply:
Another question is: can any of the numbers be 0? At `Math.log(0)` will return `-Infinity`